Media.Infrastructure/SqlRecordsRepository.cs

```diff
@@ -204,40 +204,42 @@ private static IQueryable<DBContext.Models.Records> CreateFilterQuery(Applicatio
     var filterDateQuery = tagFilter.StartDate.HasValue && tagFilter.EndDate.HasValue && tagFilter.EndDate >= tagFilter.StartDate ? $"AND (('{tagFilter.StartDate.Value.ToUniversalTime().Date:O}' <= date_trunc('day', rec.date::timestamptz, 'UTC')) AND date_trunc('day', rec.date::timestamptz, 'UTC') <= '{tagFilter.EndDate.Value.ToUniversalTime().Date:O}')" : string.Empty;
     var filterArtistsQuery = tagFilter.Artists.Count > 0 ? $"AND rec.artist_id IN ({string.Join(',', tagFilter.Artists.Select(id => string.Format("'{0}'", id)))})" : string.Empty;
     var filterGenreQuery = tagFilter.Genres.Count > 0 ? $"AND rec.genre_id IN ({string.Join(',', tagFilter.Genres.Select(id => string.Format("'{0}'", id)))})" : string.Empty;
     var filterAlbumQuery = tagFilter.Albums.Count > 0 ? $"AND rec.album_id IN ({string.Join(',', tagFilter.Albums.Select(id => string.Format("'{0}'", id)))})" : string.Empty;
     var filterLanguageQuery = tagFilter.Languages.Count > 0 ? $"AND rec.language_id IN ({string.Join(',', tagFilter.Languages.Select(id => string.Format("'{0}'", id)))})" : string.Empty;
 
     var selectQuery = @"SELECT 
-                  rec.*, a.name as artist_name, t.group_id, al.album_name as album_name, g.name as genre_name, l.name as language_name,
+                  rec.*, a.name as artist_name, al.album_name as album_name, g.name as genre_name, l.name as language_name,
                   LEAD(rec.record_id, 1) OVER(ORDER BY Cast(rec.date as Date) DESC, rec.date) as next_id, 
                   LAG(rec.record_id, 1) OVER(ORDER BY Cast(rec.date as Date) DESC, rec.date) as previous_id
                   FROM public.records AS rec
                   LEFT JOIN public.artists AS a ON rec.artist_id = a.artist_id
                   LEFT JOIN public.albums AS al ON rec.album_id = al.album_id
                   LEFT JOIN public.genres AS g ON rec.genre_id = g.genre_id
                   LEFT JOIN public.languages AS l ON rec.language_id = l.language_id
-                  LEFT JOIN (SELECT g1.groups_group_id, g1.records_record_id, g2.group_id
+                  INNER JOIN (SELECT g1.groups_group_id, g1.records_record_id
 			                       FROM public.groups_records AS g1
-			                       INNER JOIN public.groups AS g2 ON g1.groups_group_id = g2.group_id
-                            ) AS t 
-                  ON rec.record_id = t.records_record_id
-                  WHERE rec.title ILIKE {0}";
+                 ";
 
     var sb = new StringBuilder();
     sb.AppendLine(selectQuery);
     sb.AppendLine(filterGroupsQuery);
+    sb.AppendLine(@") AS t 
+               ON rec.record_id = t.records_record_id");
+    sb.AppendLine(filterQuery);
     sb.AppendLine(filterDateQuery);
     sb.AppendLine(filterArtistsQuery);
     sb.AppendLine(filterGenreQuery);
     sb.AppendLine(filterAlbumQuery);
     sb.AppendLine(filterLanguageQuery);
+    sb.AppendLine("GROUP BY rec.record_id, a.name, al.album_name, g.name, l.name");
```

Comment:
Will duplicate records! Either the join above should be LEFT JOIN as it was or the group by should only be rec.record_id!

Reply:
Use Subselect instead of inner join!

Reply:
Is the group by still necessary?